ob1k-cache/src/main/java/com/outbrain/ob1k/cache/memcache/MemcacheClient.java

```diff
@@ -63,14 +93,32 @@ private static int expirationInSpyUnits(final long exp, final TimeUnit unit) {
 
   @Override
   public ComposableFuture<V> getAsync(final K key) {
-    return SpyFutureHelper.fromGet(new SpyFutureHelper.GetFutureProducer<V>() {
+    final ComposableFuture<V> getFutureValue = SpyFutureHelper.fromGet(new SpyFutureHelper.GetFutureProducer<V>() {
       @Override
       public Future<V> createFuture() {
         @SuppressWarnings("unchecked")
         final Transcoder<V> transcoder = (Transcoder<V>) spyClient.getTranscoder();
+
         return spyClient.asyncGet(keyTranslator.translateKey(key), transcoder);
       }
     });
+
+    final Context timer = (metricFactory != null) ? reqTimer.time() : null;
```

Comment:
reject: I want to use (metricFactory != null) consistently throughout the code. In this way I can bench several operations on metrics under a single if (metricFactory != null) condition.